onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -5139,10 +5139,25 @@ def aten_square(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_squeeze(self: TensorType) -> TensorType:
+@torch_op("aten::squeeze", trace_only=True)
+def aten_squeeze(self: TTensor, dim: Optional[int] = None) -> TTensor:
     """squeeze(Tensor(a) self) -> Tensor(a)"""
 
-    raise NotImplementedError()
+    if op.OptionalHasElement(dim):
+        rank = op.Size(op.Shape(self))
+        if rank == 0:
+            self = op.Reshape(self, op.Constant(value_ints=[-1]))
+        # check if specified dimension equal to 1
+        shape = op.Shape(self, start=dim, end=dim + 1)  # type: ignore[operator]
```

Comment:
We can't use computed values like `dim + 1` as an attribute. We will need to get the shape, and use a slice op to extract something like this.

Reply:
got it. changed.